instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -110,9 +110,13 @@ def _before_request():
         return
 
     environ = flask.request.environ
-    span_name = flask.request.endpoint or otel_wsgi.get_default_span_name(
-        environ
-    )
+    span_name = None
+    try:
+        span_name = flask.request.url_rule.rule
+    except AttributeError:
+        pass
+    if span_name is None:
+        span_name = otel_wsgi.get_default_span_name(environ)
```

Comment:
is there a test that exerts this code path?

Reply:
The `test_404` test  covers this.